course/templates/course/flow-page.html

```diff
@@ -284,6 +284,55 @@
 
   {# }}} #}
 
+
+  {# {{{ flow page interaction email #}
+
+  {% if may_send_email_about_flow_page %}
+      <div class="row" align="right">
+          <a href="{% url 'relate-flow_page_interaction_email' course.identifier flow_session.id ordinal %}"
+             class="btn btn-primary btn-xs relate-btn-xs-vert-spaced"
+             title="{% trans 'Send an interaction email to course staff about this page' %}"
+             target="_blank"
+          >
+          {% trans "Send email" %}
+      </a>
+      </div>
+  {% endif %}
+
+  {# }}} #}
+
+
+  {# {{{ show single grade link #}
+
+  {% if show_privileged_shortcut %}
+    <div class="row" align="right">
+       {% if not flow_session.in_progress %}
+            <a href="{% url 'relate-view_reopen_session' course.identifier flow_session.id opportunity.id %}"
+               class="btn btn-primary btn-xs relate-btn-xs-vert-spaced"
+               target="_blank"
+            >
+              {% trans "Reopen" %}
+            </a>
+        {% endif %}
+          <a href="{% url 'relate-grant_exception_stage_3' course.identifier flow_session.participation.id flow_session.flow_id flow_session.id %}"
+             class="btn btn-primary btn-xs relate-btn-xs-vert-spaced"
+             target="_blank"
+          >
+              {% trans "Grant exception" %}
+          </a>
+
+      <a class="btn btn-primary btn-xs relate-btn-xs-vert-spaced"
+         href="{% url 'relate-grade_flow_page' course.identifier flow_session.id ordinal %}"
+         role="button"
+         target="_blank"
+      >
+        {% trans "View in grading page" %}
```

Comment:
Never say sorry about that. I really like the opportunity to improve the code. Thanks for the information.   Updated.